kapitan/initialiser.py

```diff
@@ -20,27 +20,143 @@
 import os
 import sys
 import shutil
+import yaml
+
 from distutils.dir_util import copy_tree
 
 logger = logging.getLogger(__name__)
 
-def initialise_skeleton(directory):
-    """ Initialises a directory with a recommended skeleton structure
-    Args:
-        directory (string): path which to initialise, directory is assumed to exist
-    """
-
-    current_pwd = os.path.dirname(__file__)
-    templates_directory = os.path.join(current_pwd, 'inputs', 'templates')
-
-    copy_tree(templates_directory, directory)
-
-    logger.info("Populated {} with:".format(directory))
-    for dirName, subdirList, fileList in os.walk(directory):
-        logger.info('{}'.format(dirName))
-        for fname in fileList:
-            logger.info('\t {}'.format(fname))
-        # Remove the first entry in the list of sub-directories
-        # if there are any sub-directories present
-        if len(subdirList) > 0:
-            del subdirList[0]
+class Initialiser(object):
+    def __init__(self,directory,targets,compile_inputs):
+        """ Initialises a directory with a recommended skeleton structure
+        Args:
+            directory (string): path which to initialise, directory is assumed to exist
+            targets (list): target files to be created under targets/
+            compile_inputs (list): input type of templates to be generated
+        """
+        self.target_template_name = 'my_target.yml'
+        self.component_template_name = 'my_component.yml'
+        self.target_dir = "inventory/targets"
```

Comment:
the inventory path should not be hardcoded. It should take the value of `--inventory-path`.